src/Features/CSharp/Portable/CodeLens/DisplayInfoCSharpServices.cs

```diff
@@ -0,0 +1,348 @@
+        /// </summary>
+        public bool IsDocumentationComment(SyntaxNode node)
+        {
+            return node.IsKind(SyntaxKind.SingleLineDocumentationCommentTrivia) ||
+                   node.IsKind(SyntaxKind.MultiLineDocumentationCommentTrivia);
+        }
+
+        /// <summary>
+        /// Returns the node that should be displayed
+        /// </summary>
+        public SyntaxNode GetDisplayNode(SyntaxNode node)
+        {
+            while (true)
+            {
+                switch (node.Kind())
+                {
+                    // LocalDeclarations do not have symbols themselves, you need a variable declarator
+                    case SyntaxKind.LocalDeclarationStatement:
+                        var localDeclarationNode = (LocalDeclarationStatementSyntax) node;
+                        node = localDeclarationNode.Declaration.Variables.First();
+                        continue;
+
+                        // Field and event declarations do not have symbols themselves, you need a variable declarator
+                    case SyntaxKind.FieldDeclaration:
+                    case SyntaxKind.EventFieldDeclaration:
+                        var fieldNode = (BaseFieldDeclarationSyntax) node;
+                        node = fieldNode.Declaration.Variables.First();
+                        continue;
+
+                        // Variable is a field without access modifier. Parent is FieldDeclaration
+                    case SyntaxKind.VariableDeclaration:
+                        node = node.Parent;
+                        continue;
+
+                        // Built in types   
+                    case SyntaxKind.PredefinedType:
+                        node = node.Parent;
+                        continue;
+
+                    case SyntaxKind.MultiLineDocumentationCommentTrivia:
+                    case SyntaxKind.SingleLineDocumentationCommentTrivia:
+                        // For DocumentationCommentTrivia node, node.Parent is null. Obtain parent through ParentTrivia.Token
+                        if (node.IsStructuredTrivia)
+                        {
+                            var structuredTriviaSyntax = (StructuredTriviaSyntax) node;
+                            node = structuredTriviaSyntax.ParentTrivia.Token.Parent;
+                            continue;
+                        }
+                        return null;
+
+                    default:
+                        return node;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the DisplayName for the given node.
+        /// </summary>
+        public string GetDisplayName(
+            SemanticModel semanticModel,
+            SyntaxNode node,
+            DisplayFormat displayFormat)
+        {
+            if (node == null)
+            {
+                return CSharpFeaturesResources.unknown_value;
+            }
+
+            SymbolDisplayFormat symbolDisplayFormat = DefaultFormat;
```

Comment:
Just trying to make sure I understand the coding convention since I'm relatively new to the code base. The .NET guidelines say "We only use var when it's obvious what the variable type is (i.e. var stream = new FileStream(...) not var stream = OpenStandardInput())." In this case, there's no mention of SymbolDisplayFormat on the RHS, are you saying the variable name is generally counts as making it "obvious". (R#, for example, does not flag this as "var allowed where evident.") Just want to make sure I'm clear for future reviews...

---
In reply to: [73584198](https://github.com/dotnet/roslyn/pull/12930#discussion_r73584198) [](ancestors = 73584198)